blackmarket/Managers/SessionManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SessionDataManager.swift
+//  ios-base
+//
+//  Created by Juan Pablo Mazza on 11/8/16.
+//  Copyright © 2016 Rootstrap Inc. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+import Combine
+
+internal class SessionManager: CurrentUserSessionProvider {
+    
+    static let Authenticated = PassthroughSubject<Bool, Never>()
+    
+    static func IsAuthenticated() -> Bool {
+        return SessionManager.shared.validSession
+    }
```

Comment:
Please use `validSession`, having a static variable is not desirable since you will be using an instance of SessionManager at all times.

```suggestion
```